src/parsers/manifest/dash/wasm-parser/rs/events.rs

```diff
@@ -0,0 +1,361 @@
+    TimeScale = 27, // f64
+
+    // SegmentURL + SegmentTemplate
+    Index = 28, // String
+
+    // Initialization
+    InitializationRange = 29, // [f64, f64]
+
+    // SegmentURL + SegmentTemplate + SegmentBase + Initialization
+    Media = 30, // String
+    IndexRange = 31, // [f64, f64]
+
+    // Period + AdaptationSet + SegmentTemplate
+    BitstreamSwitching = 32, // u8 (bool)
+
+
+    // MPD
+    Type = 33, // String
+    AvailabilityStartTime = 34, // f64
+    AvailabilityEndTime = 35, // f64
+    PublishTime = 36, // f64
+    MinimumUpdatePeriod = 37, // f64
+    MinBufferTime = 38, // f64
+    TimeShiftBufferDepth = 39, // f64
+    SuggestedPresentationDelay = 40, // f64
+    MaxSegmentDuration = 41, // f64
+    MaxSubsegmentDuration = 42, // f64
+
+    // BaseURL + SegmentTemplate
+    AvailabilityTimeOffset = 43, // f64
+
+    // BaseURL
+    BaseUrlValue = 44, // String
+
+    // Period
+    Start = 45, // f64
+    XLinkHref = 46, // String
+    XLinkActuate = 47, // String
+
+    // AdaptationSet
+    Group = 48,
+    MaxBandwidth = 49, // f64
+    MaxFrameRate = 50, // f64
+    MaxHeight = 51, // f64
+    MaxWidth = 52, // f64
+    MinBandwidth = 53, // f64
+    MinFrameRate = 54, // f64
+    MinHeight = 55, // f64
+    MinWidth = 56, // f64
+    SelectionPriority = 57,
+    SegmentAlignment = 58,
+    SubsegmentAlignment = 59,
+
+    // AdaptationSet + ContentComponent
+    Language = 60, // String
+    ContentType = 61, // String
+    Par = 62,
+
+    // Representation
+    Bitrate = 63, // f64
+
+    Text = 64,
+    QualityRanking = 65,
+    Location = 66,
+
+    InitializationMedia = 67,
+
+    /// Describes an encountered "mediaPresentationDuration" attribute, as found
+    /// in `<MPD>` elements.
+    ///
+    /// This value has been converted into seconds, as an f64.
+    MediaPresentationDuration = 68,
+
+    /// Describes the byte range (end not included) of an encountered `<Event>`
+    /// element in the whole MPD.
+    ///
+    /// This can be useful to re-construct the whole element on the JS-sid.
+    ///
+    /// It is reported as an array of two f64 values.
+    /// The first number indicating the starting range (included).
+    /// The second indicating the ending range (non-included).
+    EventStreamEltRange = 69,
+}
+
+impl TagName {
+    /// Signal a new tag opening to the application
+    pub fn report_tag_open(self) {
+        debug_assert!(self as u64 <= u8::MAX as u64);
+
+        // UNSAFE: We're using FFI, but there should be no risk at all here
+        unsafe { onTagOpen(self) };
+    }
+
+    /// Signal that a previously-open tag closed to the application
+    pub fn report_tag_close(self) {
+        debug_assert!(self as u64 <= u8::MAX as u64);
+
+        // UNSAFE: We're using FFI, but there should be no risk at all here
+        unsafe { onTagClose(self) };
+    }
+}
+
+use crate::reportable::ReportableAttribute;
+use crate::utils::*;
```

Comment:
Here, is it necessary to 'use' the whole content of utils ? Or would it be better to 'use' only function, etc that you need here ?

I don't know Rust, it is more of a curious question than a note :)

Reply:
done